vault/logical_system_paths.go

```diff
@@ -64,38 +84,208 @@ func (b *SystemBackend) configPaths() []*framework.Path {
+					},
+				},
+			},
+		},
 
 		{
-			Pattern:         "init$",
+			Pattern: "init$",
+			Fields: map[string]*framework.FieldSchema{
+				"pgp_keys": &framework.FieldSchema{
+					Type:        framework.TypeCommaStringSlice,
+					Description: "Specifies an array of PGP public keys used to encrypt the output unseal keys. Ordering is preserved. The keys must be base64-encoded from their original binary representation. The size of this array must be the same as secret_shares.",
+				},
+				"root_token_pgp_key": &framework.FieldSchema{
+					Type:        framework.TypeString,
+					Description: "Specifies a PGP public key used to encrypt the initial root token. The key must be base64-encoded from its original binary representation.",
+				},
+				"secret_shares": &framework.FieldSchema{
+					Type:        framework.TypeInt,
+					Description: "Specifies the number of shares to split the master key into.",
+				},
+				"secret_threshold": &framework.FieldSchema{
+					Type:        framework.TypeInt,
+					Description: "Specifies the number of shares required to reconstruct the master key. This must be less than or equal secret_shares. If using Vault HSM with auto-unsealing, this value must be the same as secret_shares.",
+				},
+				"stored_shares": &framework.FieldSchema{
+					Type:        framework.TypeInt,
+					Description: "Specifies the number of shares that should be encrypted by the HSM and stored for auto-unsealing. Currently must be the same as secret_shares.",
+				},
+				"recovery_pgp_keys": &framework.FieldSchema{
+					Type:        framework.TypeCommaStringSlice,
+					Description: "Specifies an array of PGP public keys used to encrypt the output recovery keys. Ordering is preserved. The keys must be base64-encoded from their original binary representation. The size of this array must be the same as recovery_shares.",
+				},
+			},
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.ReadOperation: &framework.PathOperation{
+					Summary: "Returns the initialization status of Vault.",
+				},
+				logical.CreateOperation: &framework.PathOperation{
+					Summary:     "Initialize a new Vault.",
+					Description: "The Vault must not have been previously initialized. The recovery options, as well as the stored shares option, are only available when using Vault HSM.",
+				},
+			},
+
 			HelpSynopsis:    strings.TrimSpace(sysHelp["init"][0]),
 			HelpDescription: strings.TrimSpace(sysHelp["init"][1]),
 		},
+		{
+			Pattern: "leader$",
+
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.ReadOperation: &framework.PathOperation{
+					Summary: "Returns the high availability status and current leader instance of Vault.",
+				},
+			},
+
+			HelpSynopsis: "Check the high availability status and current leader of Vault",
+		},
+		{
+			Pattern: "step-down$",
+
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.CreateOperation: &framework.PathOperation{
```

Comment:
Shouldn't this (and elsewhere) be UpdateOperation? Anywhere we don't have both Create and Update, the router expects to find an Update operation.

Reply:
Yep. These weren't causing any failures b/c the router doesn't use them, but they wouldn't work in normal use and I've now fixed them.